src/hardware/voodoo.cpp

```diff
@@ -7060,46 +7060,43 @@ static constexpr uint32_t voodoo_r(const uint32_t addr)
 
 static int get_num_threads()
 {
-	auto get_auto_threads = []() {
-		// Sweet spot found to be seven threads based on benchmarks:
-		// https://github.com/schellingb/dosbox-pure/issues/300#issuecomment-2144050898
-		constexpr auto MinThreads = 0;
-		constexpr auto MaxThreads = 7;
-		return std::clamp(SDL_GetCPUCount() - 1, MinThreads, MaxThreads);
-	};
-
-	auto parse_positive_base10_int = [](const std::string& str) {
-		constexpr auto Base10 = 10;
-		const auto maybe_int  = parse_int(str, Base10);
-		return (maybe_int && *maybe_int >= 0) ? maybe_int : std::nullopt;
-	};
-
-	// Get the user's "[voodoo] voodoo_threads" setting
-	constexpr auto DefaultSetting = "auto";
+	constexpr auto MinThreads     = 0;
+	constexpr auto MaxAutoThreads = 7;
+	constexpr auto MaxThreads     = 16;
+	constexpr auto AutoSetting    = "auto";
+
 	const auto sec = dynamic_cast<Section_prop*>(control->GetSection("voodoo"));
 	const auto user_setting = sec ? sec->Get_string("voodoo_threads")
-	                              : DefaultSetting;
-
-	// Did the user provide the default or a standardized positive setting?
-	if (user_setting == DefaultSetting || has_true(user_setting)) {
-		return get_auto_threads();
-	}
-	// Did the user provide a custom value?
-	if (const auto maybe_int = parse_positive_base10_int(user_setting)) {
-		return *maybe_int;
-	}
-	// Did the user provide a standardized negative setting?
-	if (has_false(user_setting)) {
-		return 0;
+	                              : AutoSetting;
+
+	if (const auto maybe_int = parse_int(user_setting)) {
+		const auto valid_int = std::clamp(*maybe_int, MinThreads, MaxThreads);
+
+		if (*maybe_int != valid_int) {
+			LOG_WARNING(
+			        "VOODOO: Invalid custom 'voodoo_threads' quantity: '%s', "
+			        "Valid values range from '%d' to '%d'. Using '%d'",
```

Comment:
Minor suggestion: maybe we can follow the wording of the generic out-of-range error message of pure integer conf settings.

<img width="1085" alt="image" src="https://github.com/user-attachments/assets/8c69dad9-3b8e-4f80-afaa-599346627140">

Reply:
Will do!